fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java

```diff
@@ -257,14 +262,37 @@ private static TFetchSchemaTableDataResult resourceGroupsMetadataResult(TMetadat
             trow.addToColumnValue(new TCell().setStringVal(rGroupsInfo.get(1)));
             trow.addToColumnValue(new TCell().setStringVal(rGroupsInfo.get(2)));
             trow.addToColumnValue(new TCell().setIntVal(value));
-            dataBatch.add(trow);
+            TRow filterRow = filterColumns(trow, params.getColumnsName(), TMetadataType.RESOURCE_GROUPS);
```

Comment:
I think we can use a more easy way to implement this. To avoid copy duplicate code for each TVF.

Reply:
done